### DISCUSSION TITLE
Feature Request: Hide panel query (SQL text) from view role user or hide whole JSON

### DISCUSSION THREAD

**Discussion by baur** [grafana/grafana#38311] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

Hi, 
Please, hide panel query (SQL) from view role user or hide whole JSON.

![image](https://user-images.githubusercontent.com/2898361/122854782-0de8fe00-d336-11eb-953a-2a53b20410e6.png)

![image](https://user-images.githubusercontent.com/2898361/122854518-a763e000-d335-11eb-896b-bd11e611e6f1.png)

**Why is this needed**:

For security reason

**Comment by baur** [comment]
and hide library panel too (view role user)

![image](https://user-images.githubusercontent.com/2898361/122855334-e8a8bf80-d336-11eb-8d1f-f5948aae257e.png)

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
